libs/live-nft-react/src/NftMetadataProvider.tsx

```diff
@@ -117,6 +117,46 @@ export function useNftCollectionMetadata(
   }
 }
 
+export function useNftCollectionMetadataBatch(
+  items: Array<{
+    contract: string | undefined;
+    currencyId: string | undefined;
+  }>,
+): Array<NFTResource<NonNullable<NFTCollectionMetadataResponse["result"]>>> {
+  const { cache, loadCollectionMetadata } = useContext(NftMetadataContext);
+
+  const hasAllProperties = (item: { contract?: string; currencyId?: string }) =>
+    item.contract && item.currencyId;
+
+  const data = items.map(item => {
```

Comment:
the map will create a new object on each render which invalidate the useEffect dependencies. I think it should be wrapped in a `useMemo(..., [items])`